extensions/tika/runtime/src/main/java/io/quarkus/tika/TikaParser.java

```diff
@@ -67,9 +115,28 @@ protected TikaContent parseStream(InputStream entityStream, String contentType,
         }
     }
 
-    private Parser createParser() {
-        // TODO Use the properties to support loading the format-specific and other generic parsers
-        return new AutoDetectParser();
+    private ContentHandler validateContentHandler(ContentHandler contentHandler) {
+        if (!this.appendEmbeddedContent && !(contentHandler instanceof RecursiveParserWrapperHandler)) {
+            throw new IllegalStateException(
+                    "The master and every embedded document will require a unique ContentHandler instance");
+        }
+        return contentHandler;
+    }
+
+    private ContentHandler createContentHandler() {
+        return this.appendEmbeddedContent ? new ToTextContentHandler()
+                : new RecursiveParserWrapperHandler(
+                        new BasicContentHandlerFactory(BasicContentHandlerFactory.HANDLER_TYPE.TEXT, -1));
+    }
+
+    private ContentHandler createContentHandlerForMetadataOnly(String contentType) {
+        if (this.appendEmbeddedContent) {
+            return contentType != null && contentType.contains("pdf") ? null : new DefaultHandler();
```

Comment:
I'm not sure I understand why pdf is special cased?

Reply:
I'm fond of this check because it is my only contribution to the PDF parsing story in Tika :-). Typically, when one needs the metadata only, the parsers are still processing and reporting the text which is not needed, ideally. But as I was explained, some formats have the metadata mixed in with the text, etc, and it is impossible to avoid parsing the text. Luckily, for the PDF format, it is possible, by providing a null handler, completely avoid the text processing work.

Reply:
I'll add the comment to the code